packages/admin/admin/src/dataGrid/excelExport/DataGridExcelExportButton.tsx

```diff
@@ -0,0 +1,44 @@
+import { Button, CircularProgress } from "@mui/material";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { FileIcon } from "../../fileIcons/FileIcon";
+import { ExportApi } from "./useDataGridExcelExport";
+
+interface Props {
+    exportApi: ExportApi;
+    onClick?: () => void;
+    loadingComponent?: React.ReactNode;
+    onError?: () => void;
+}
+
+export const DataGridExcelExportButton: React.FunctionComponent<Props> = ({ onClick, children, exportApi, loadingComponent, onError }) => {
+    const onClickButtonPressed = () => {
+        if (onClick != null) {
+            onClick();
+        } else {
+            exportApi.exportGrid();
+        }
+    };
+
+    const { loading, error } = exportApi;
+
+    if (error && onError) {
+        onError();
+    }
```

Comment:
I find this API strange. How would I use this `onError` in practice? Would I use it to set a state in another component?

Also, I believe since the method is called directly in the render method, this might lead to multiple calls of `onError` in some constellation.

I'd prefer it if the button would handle the error on its own. You could do that by turning it red and showing an error icon as startIcon for 3 seconds. 
In the future we will have a FeedbackButton for this, but sadly it's still in development.